src/rules.c

```diff
@@ -86,9 +86,8 @@ int read_rules() {
     } else {
 	showstring("There is no contest rules file:", contest_conf);
 	showmsg("Assuming regular QSO operation. Logfile is qso.log");
-	strcpy(whichcontest, "qso");	/* default use general qso mode...
+	setcontest("qso");		 /* default use general qso mode...
```

Comment:
"qso" is in fact config_qso.name

Reply:
Right. The idea behind setcontest(char *name) is to activate the contest (in these case the non-contest qso-mode) by name.

Reply:
I mean the string constant could be replaced by a reference to the actual source of definition.

Reply:
Yes, I understood. It seemed only weird to me as we lookup configurations by name and then use that waht we would fine to look itself up.

But maybe you are right. Atm the best it seems to define a symbol and use it in both places - as a name for the configuration and as a name to look it up

#define QSO_MODE "qso"

config_qso = { .name = QSO_MODE }  in setcontest.c

and finally

setcontest(QSO_MODE)                         in rules.c